subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Output directories inside build dir
+set(src_out ${ZEPHYR_BINARY_DIR}/source/generated)
+set(include_out ${ZEPHYR_BINARY_DIR}/include/generated)
+
+# Make sure that output directory for *.c files exist
+file(MAKE_DIRECTORY ${src_out})
+
+# This file is used as the source for the parser generator
+set(cddl_file ${CMAKE_CURRENT_LIST_DIR}/modem_update.cddl)
+
+# These are the entry types needed by the source code
+set(entry_types Wrapper Sig_structure1 Segments)
+
+set(decode_c_name modem_update_decode.c)
+set(decode_h_name modem_update_decode.h)
+set(types_h_name modem_update_types.h)
+set(decode_c ${src_out}/${decode_c_name})
+set(decode_h ${include_out}/${decode_h_name})
+set(types_h ${include_out}/${types_h_name})
+set(src_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
+set(license ${CMAKE_CURRENT_LIST_DIR}/license.cmake)
+
+# The .clang-format in this repo states support for >= version 4
+find_program(
+  CLANG_FORMAT
+  NAMES
+  clang-format
+  clang-format-10
+  clang-format-9
+  clang-format-8
+  clang-format-7
+  clang-format-6
+  clang-format-5
+  clang-format-4
+  )
+
+if (${CLANG_FORMAT} STREQUAL CLANG_FORMAT-NOTFOUND)
+  message(WARNING
+    "'clang-format' not found, generated code will not be formatted")
+else()
+  set(CLANG_FORMAT_COMMAND
+   COMMAND ${CLANG_FORMAT} -i "${decode_c};${decode_h};${types_h}")
+endif()
+
+add_custom_command(
+  OUTPUT ${decode_c} ${decode_h} ${types_h}
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_CDDL_GEN_MODULE_DIR}/scripts/cddl_gen.py
```

Comment:
This requires `regex` to be added to requirements.txt.

I think https://github.com/nrfconnect/sdk-nrf/blob/master/scripts/requirements-build.txt is the right one to use.